CMakeLists.txt

```diff
@@ -115,7 +115,16 @@ tristate_option(WITH_USDT
 )
 cmake_dependent_option(WITH_EXTERNAL_SIGNER "Enable external signer support." ON "NOT WIN32" OFF)
 set(ENABLE_EXTERNAL_SIGNER ${WITH_EXTERNAL_SIGNER})
-tristate_option(WITH_QRENCODE "Enable QR code support." "if libqrencode is found." AUTO)
+
+include(cmake/optional_qt.cmake)
+cmake_dependent_option(WITH_QRENCODE "Enable QR code support." ON "WITH_GUI" OFF)
+if(WITH_QRENCODE)
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(libqrencode REQUIRED IMPORTED_TARGET libqrencode)
```

Comment:
Only tangentially related to this PR, but the error printed by `pkg_check_modules` is kind of ugly for a user facing message:
```
CMake Error at /usr/share/cmake-3.28/Modules/FindPkgConfig.cmake:619 (message):
  The following required packages were not found:

   - libqrencode

Call Stack (most recent call first):
  /usr/share/cmake-3.28/Modules/FindPkgConfig.cmake:841 (_pkg_check_modules_internal)
  CMakeLists.txt:123 (pkg_check_modules)
```
Would be nice to have something along the lines of what configure would print in a similar case:
```
configure: error: QR support requested but cannot be built. Use --without-qrencode
```